src/generators/non_gradient_based/growing_spheres/growing_spheres.jl

```diff
@@ -38,70 +38,45 @@ function growing_spheres_generation!(ce::AbstractCounterfactualExplanation)
     factual = ce.x
     counterfactual_data = ce.data
     target = [ce.target]
-    max_iter = 1000
 
     # Copy hyperparameters
     n = generator.n
     η = convert(eltype(factual), generator.η)
 
     # Generate random points uniformly on a sphere
     counterfactual_candidates = hyper_sphere_coordinates(n, factual, 0.0, η)
-
-    if (factual == target)
-        ce.s′ = factual
-        return nothing
-    end
+    update_path(ce, counterfactual_candidates)
 
     # Predict labels for each candidate counterfactual
     counterfactual = find_counterfactual(
         model, target, counterfactual_data, counterfactual_candidates
     )
 
     # Repeat until there's no counterfactual points (process of removing all counterfactuals by reducing the search space)
-    while (!isnothing(counterfactual))
+    while (!isnothing(counterfactual) && ce.search[:iteration_count] < ce.convergence[:max_iter])
         η /= 2
         a₀ = convert(eltype(factual), 0.0)
 
         counterfactual_candidates = hyper_sphere_coordinates(n, factual, a₀, η)
+        update_path(ce, counterfactual_candidates)
```

Comment:
Now we're updating path every time we consider a new candidate for a counterfactual